src/Chapter3.hs

```diff
@@ -970,6 +1087,28 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Enum, Show)
+
+isWeekend :: WeekDay -> Bool
+isWeekend Sunday = True
+isWeekend Saturday = True
+isWeekend _ = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sunday = Monday
+nextDay day = succ day
+
+daysToParty :: WeekDay -> Int
+daysToParty Friday = 0
+daysToParty x = 1 + daysToParty (nextDay x)
```

Comment:
That's the correct solution :+1: But you make it slightly more efficient by using the `fromEnum` method of the `Enum` typeclass :slightly_smiling_face: